src/pieces/Knight.js

```diff
@@ -8,6 +8,26 @@ class Knight extends Piece {
   }
   findLegalMoves() {
     const possibleMoves = [];
+    if (this.side == 'white' && this.y - 2 >= 0 && this.x - 1 >= 0 && this.y - 2 <= 7 && this.x - 1 <= 7) possibleMoves.push(`${this.x - 1},${this.y - 2}`);
+    if (this.side == 'white' && this.y - 2 >= 0 && this.x + 1 >= 0 && this.y - 2 <= 7 && this.x + 1 <= 7) possibleMoves.push(`${this.x + 1},${this.y - 2}`);
+    if (this.side == 'white' && this.y - 1 >= 0 && this.x - 2 >= 0 && this.y - 1 <= 7 && this.x - 2 <= 7) possibleMoves.push(`${this.x - 2},${this.y - 1}`);
+    if (this.side == 'white' && this.y + 1 >= 0 && this.x - 2 >= 0 && this.y + 1 <= 7 && this.x - 2 <= 7) possibleMoves.push(`${this.x - 2},${this.y + 1}`);
+    if (this.side == 'white' && this.y - 1 >= 0 && this.x + 2 >= 0 && this.y - 1 <= 7 && this.x + 2 <= 7) possibleMoves.push(`${this.x + 2},${this.y - 1}`);
+    if (this.side == 'white' && this.y + 1 >= 0 && this.x + 2 >= 0 && this.y + 1 <= 7 && this.x + 2 <= 7) possibleMoves.push(`${this.x + 2},${this.y + 1}`);
+    if (this.side == 'white' && this.y + 2 >= 0 && this.x - 1 >= 0 && this.y + 2 <= 7 && this.x - 1 <= 7) possibleMoves.push(`${this.x - 1},${this.y + 2}`);
+    if (this.side == 'white' && this.y + 2 >= 0 && this.x + 1 >= 0 && this.y + 2 <= 7 && this.x + 1 <= 7) possibleMoves.push(`${this.x + 1},${this.y + 2}`);
```

Comment:
Sugeruje pomyśleć chwilkę i spróbować zamknąć to w jakaś pętle.

Można na przykład zrobić lokalnie funkcję która sprawdzi czy dane koordynaty zawierają się wewnątrz gry ( czyli czy liczby x i y należą do przedziału <0,7> ). Następnie zakodować tablice wszystkich kierunków w jakie może się skoczek ruszyć (przykładowo tablica [2,1] to dwa w dół jeden w prawo, a  [-1,-2] to jeden w góre i dwa w lewo). Potem wystarczy przepuścić aktualne położenie przez wszystkie możliwe kierunki i jeżeli funkcja sprawdzająca na to pozwoli to do rzucić do tablicy.

Polecam spróbować, jak się uda to spoko jak nie to zostaniemy przy tych ifach